app/utils/service_client/middleware/jwt_authenticator.rb

```diff
@@ -1,21 +1,24 @@
 module ServiceClient
   module Middleware
 
+    IDENTITY = ->() {}
+
     AuthContext = EntityUtils.define_builder(
       [:marketplace_id, :uuid, :mandatory],
       [:actor_id, :uuid, :mandatory]
     )
 
     class JwtAuthenticator < MiddlewareBase
 
-      def initialize(disable_authentication, token_secret)
-        @_disabled = disable_authentication
-        @_secret = token_secret
+      def initialize(disable:, secret:, default_auth_context: IDENTITY)
+        @_disabled = disable
+        @_secret = secret
+        @_default_auth_context = default_auth_context
       end
 
       def enter(ctx)
         unless @_disabled
-          token = create_token(ctx[:opts][:auth_context])
+          token = create_token(ctx[:opts][:auth_context] || @_default_auth_context.call)
```

Comment:
I guess you mean "in case authentication is **enabled**", right?

If that happens, the Entity call on [line 30](https://github.com/sharetribe/sharetribe/pull/2733/files/cacc9bcc594dd584de87964c73c720f93567b521#diff-37cd77e957ff88d796c4f822409c28e8R30) will throw an error because mandatory values are missing.